source/includes/inkling2-reference/_functions.html.md

```diff
@@ -0,0 +1,233 @@
+
+
+### Function Statements
+
+#### Variable Declarations
+
+Variables can be declared within a function. All variables have function-level scope even if they are declared within a sub-scope such as an if/else statement. Each variable within a function must have a unique name and a type. The type can be omitted if it can be inferred from an initial value assigned to the variable.
+
+```inkling2--syntax
+variableDeclaration :==
+  var <varName> [':' <varType>] ['=' <expression>]
+```
+
+A variable that is not assigned an initial value must be assigned a value along all code paths before it is used.
+
+```inkling2--code
+# Variable with no initial assignment.
+var TwoOrThree: number<2, 3>
+
+if IsCondition() {
+  TwoOrThree = 2
+}
+
+# This will generate a compiler error because the
+# variable may not have a defined value.
+return TwoOrThree
+```
+
+#### Variable Assignments
+A previously-declared variable can be assigned a new value as long as that value matches the type of the variable.
+
+```inkling2--syntax
+variableAssignment :==
+  <varName> '=' <expression>
+```
+
+#### Conditional Execution
+Inkling supports if/else statements, which can be chained together with “else if” clauses. The braces that surround the code blocks within an `if` or `else` clause are required.
+
+```inkling2--syntax
+conditionalStatement :==
+  if <expression> '{'
+    [functionStatement]*
+  '}' [else if <expression> '{'
+    [functionStatement]*
+  '}']* [else '{'
+    [functionStatement]*
+  '}']
+```   
+
+#### Return Statements
+A return statement terminates execution of the function and returns the specified value to its caller. The type of the expression must match the declared return type of the function.
+
+```inkling2--syntax
+returnStatement :==
+  return <expression>
+```
+
+### Expressions
+Inkling supports basic expressions and a variety of unary and binary operators.
+
+#### Built-in Operators
+| Operation        | Syntax        | Description |
+| -                | -             | -           |
+| Function call    | X()           | Invokes a function with zero or more arguments |
+| Array index      | X[Y]          | Indexes into an array |
+| Member access    | X.Y           | References a field within a structure |
+| Neg/pos          | -, +          | Negate numeric value |
+| Logical not      | not           | Logical not (produces 0 or 1) |
+| Exponent         | **            | X raised to the Y power |
+| Mult/div/mod     | *, /, %       | X times Y, X divided by Y, X mod Y |
+| Add/subtract     | +, -          | X plus Y, X minus Y |
+| Relational       | <, <=, >, >=  | Less, less-or-equal, greater, greater-or-equal |
+| Equality         | ==, !=        | Equal, not equal |
+| Logical and      | and           | Left expression if truthy, else right expression |
```

Comment:
"truthy" is not defined. Is 0 the only false value?

Reply:
Truthy is a common term used in programming languages. Those people who care about this detail can look up the term if they're not familiar with it. Logical binary operators "or" and "and" do not return "true" or "false" in Inkling or in most languages (Javascript, Python, etc.). They return either the value of the left or right operand.

Reply:
I suggested writing it out because I'm not sure our target users will know it -- they may have been coding for years, but probably in Matlab or Fortran or C, and typically don't get into programming terminology -- just figure out how get what they need done. Also, our rules are different than both js and python, which are different from each other. e.g. we're not allowing `if [] {...}`.